File: frontend/src/services/payroll.ts
import http from '@/services/http'
import type { Paged } from '@/services/organization'

export interface EmployeeSalary {
  employeeId: number
  employeeName: string | null
  employeeNo: string
  baseSalary: number
  positionAllowance: number
  mealAllowance: number
  effectiveDate: string
  updatedAt: string
}

export interface UpsertEmployeeSalary {
  baseSalary: number
  positionAllowance: number
  mealAllowance: number
  effectiveDate: string
}

export interface Payroll {
  id: number
  period: number
  employeeId: number
  employeeName: string | null
  employeeNo: string
  baseAmount: number
  overtimePay: number
  bonus: number
  leaveDeduction: number
  insuranceDeduction: number
  taxWithheld: number
  grossPay: number
  netPay: number
  status: string
  generatedBy: number
  generatedByName: string | null
  generatedAt: string
  confirmedAt: string | null
  paidAt: string | null
}

export interface GenerateResult {
  period: number
  generated: number
  skipped: number
}

export interface PayrollQuery {
  page?: number
  pageSize?: number
  period?: number
  employeeId?: number
}

export async function getPayrolls(query: PayrollQuery = {}): Promise<Paged<Payroll>> {
  const { data } = await http.get<Paged<Payroll>>('/payrolls', { params: query })
  return data
}

export async function getPayroll(id: number): Promise<Payroll> {
  const { data } = await http.get<Payroll>(`/payrolls/${id}`)
  return data
}

export async function generatePayroll(period: number): Promise<GenerateResult> {
  const { data } = await http.post<GenerateResult>('/payrolls/generate', { period })
  return data
}

export async function confirmPayroll(id: number): Promise<boolean> {
  const { data } = await http.put<boolean>(`/payrolls/${id}/confirm`)
  return data
}

export async function payPayroll(id: number): Promise<boolean> {
  const { data } = await http.put<boolean>(`/payrolls/${id}/pay`)
  return data
}

export async function setPayrollBonus(id: number, amount: number): Promise<boolean> {
  const { data } = await http.put<boolean>(`/payrolls/${id}/bonus`, { amount })
  return data
}

export async function getSalaries(): Promise<Paged<EmployeeSalary>> {
  const { data } = await http.get<Paged<EmployeeSalary>>('/employee-salaries', { params: { page: 1, pageSize: 200 } })
  return data
}

export async function getSalary(employeeId: number): Promise<EmployeeSalary> {
  const { data } = await http.get<EmployeeSalary>(`/employee-salaries/${employeeId}`)
  return data
}

export async function upsertSalary(employeeId: number, body: UpsertEmployeeSalary): Promise<EmployeeSalary> {
  const { data } = await http.put<EmployeeSalary>(`/employee-salaries/${employeeId}`, body)
  return data
}

File: backend/HrSystem.Api/Models/Payroll.cs
namespace HrSystem.Api.Models;

public class EmployeeSalary
{
    public long Id { get; set; }
    public long EmployeeId { get; set; }
    public decimal BaseSalary { get; set; }
    public decimal PositionAllowance { get; set; }
    public decimal MealAllowance { get; set; }
    public DateOnly EffectiveDate { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;
    public Employee? Employee { get; set; }
}

public class Payroll
{
    public long Id { get; set; }
    public int Period { get; set; }
    public long EmployeeId { get; set; }
    public decimal BaseAmount { get; set; }
    public decimal OvertimePay { get; set; }
    public decimal Bonus { get; set; }
    public decimal LeaveDeduction { get; set; }
    public decimal InsuranceDeduction { get; set; }
    public decimal TaxWithheld { get; set; }
    public decimal GrossPay { get; set; }
    public decimal NetPay { get; set; }
    public string Status { get; set; } = "draft";
    public long GeneratedBy { get; set; }
    public DateTimeOffset GeneratedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset? ConfirmedAt { get; set; }
    public DateTimeOffset? PaidAt { get; set; }
    public Employee? Employee { get; set; }
    public User? GeneratedByUser { get; set; }
}

public class PayrollSetting
{
    public long Id { get; set; }
    public string Code { get; set; } = string.Empty;
    public string Label { get; set; } = string.Empty;
    public decimal Value { get; set; }
    public bool IsActive { get; set; } = true;
}